### PR TITLE
Python 3 Support

### PR DESCRIPTION
**Pull request by bemcdonnell**
I've got a first iteration here for Py34 support on Windows.  If you guys could check out my branch and do some testing, that would be nice. 

This covers #80

### CODE REVIEW

**Comment by bemcdonnell** [comment]
@michaeltryby, I'm finally coming out of the stone ages ;-)

**Comment by bemcdonnell** [comment]
If you're interested here are some references:

http://stackoverflow.com/questions/7237133/create-string-buffer-throwing-error-typeerror-str-bytes-expected-instead-of-str 

http://stackoverflow.com/questions/29578469/how-to-make-an-object-both-a-python2-and-python3-iterator

**Comment by goanpeca** [comment]
@bemcdonnell why 3.4? we should aim for either 3.5 or 3.6

**Comment by goanpeca** [comment]
The changes you made to utf8 encode are not robust enough if the plan is to support both python2.7 and 3. Take a look at six library and use that please.

**Comment by bemcdonnell** [comment]
@goanpeca, I updated the encoding behavior to use the python six module.  Also, testing for 2.7, 3.4, 3.5, and 3.6 are all a pass! Let me know if you approve of these changes.

**Comment by goanpeca** [comment]
Hey @bemcdonnell looking much better

Still not sure about the `next = __next__`

File: pyswmm/links.py [hunk 1]
```diff
@@ -94,14 +94,16 @@ def __getitem__(self, linkid):
...
-    def next(self):
+    def __next__(self):
         if self._cuindex < self._nLinks:
             linkobject = Link(self._model, self._linkid)
             self._cuindex += 1  # Next Iteration
             return linkobject
         else:
             raise StopIteration()
 
+    next = __next__  # Python 2
```
Comment (goanpeca) [inline pyswmm/links.py]: Why do we need this?

**Comment by goanpeca** [comment]
@bemcdonnell we need to update the dependencies on setup.py to include six now.

File: pyswmm/links.py (hunk 1 above)
Comment (bemcdonnell) [inline pyswmm/links.py]: So as I understand, the method `__next__` was day lighted in Python 3 [pep3114](https://www.python.org/dev/peps/pep-3114/).  In py2, by default, the iterator calls `next()` therefore, I had to add this line in order for it to step.  

I could use the `2to3` package but that doesn't feel like progress to me.  If you know of a better way, please enlighten me!

Comment (goanpeca) [inline pyswmm/links.py]: `six.next(<iterator>)` ?

Comment (goanpeca) [inline pyswmm/links.py]: I mean remove those redefinitions and use directly `six.next` does not that work?

Comment (bemcdonnell) [inline pyswmm/links.py]: I'm not having any success with your suggestion.  This code porting guide shows the iterator `next` implementation as I have it above: [Conservative Python 3 Porting Guide](http://portingguide.readthedocs.io/en/latest/iterators.html#fixer-considerations) (see the end).  Maybe there is still a better way?

Comment (goanpeca) [inline pyswmm/links.py]: Ok yes you are right :-)

**Comment by bemcdonnell** [comment]
@goanpeca, shall I merge?

**Comment by goanpeca** [comment]
yeah ;-)

**Pull request merged**